### PR TITLE
Mileston 2

### PR DESCRIPTION
**Pull request by AbigiyaTY**
### In this milestone I:
- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.
- By default new tasks is set the property `completed` set to `false` and the property `index` set to the value of the new array length.
- Deleting a task is updating all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To-Do List are saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,29 +1,153 @@
...
-  description: 'complet To Do list project',
-  complete: false,
-  index: 1,
-},
-];
+const textInput = document.querySelector('input');
+const listToDo = document.querySelector('.listToDo');
+class MyObject {
+  constructor(description, completed, index) {
+    this.description = description;
```
Comment (elmar8287) [inline src/index.js:7]: - [ ] Please, consider fixing the following requirement:

- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

In your case, the indexes are not updating. Please, have a look at the following screenshots.

> This is I added 3 tasks, and you see the indexes are correct in the beginning:
Screen Shot 2022-08-19 at 09 30 53

> After, I deleted the first task and the indexes did not update:
Screen Shot 2022-08-19 at 09 31 22

File: src/index.js [hunk 2]
```diff
@@ -1,29 +1,153 @@
...
-  listArray.forEach((task) => {
-    listToDo.innerHTML
-            += `<ul id="item-list" class="items">
-                        <li class="toDo">
-                            <button class="completed">□</button>
-                            <div class="discription">${task.description}</div>
-                            <div class="dots">⋮</div>
-                        </li>
-                    </ul>`;
+// showing data in the browser by bring it from local storage
```
Comment (elmar8287) [inline src/index.js]: - [ ] Please, consider removing the liner disables from your code - let linter makes it a direct job - help you to find the bugs to fix them in the early stages of the development 👍 
- [ ] Also, I suggest you remove the comments, to make your code clear and professional. Your code is enough understandable and talks by itself about what is happening 👍

**Review by elmar8287 — CHANGES_REQUESTED** [review]
Hi, @AbigiyaTY.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good PR description 👍 
- Good README file 👍 You have also deployed the project 🥇 

## Required Changes ♻️

- [ ] Check the comments under the review.

## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @AbigiyaTY, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good implementation of changes required ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**